### PR TITLE
#5463 - feat: adding ResultItem story, styling and some refactoring.

### PR DESCRIPTION
**Pull request by DipakShrestha18**

### CODE REVIEW

File: src/components/ResultItem/ResultItem.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+type ResultItemProps = {
+  className?: string;
+  meta?: string;
+  title: string;
+  href: string;
+  text?: string;
```
Comment (chriddell) [inline src/components/ResultItem/ResultItem.tsx]: Can we do these in alphabetical order please? Just to stay consistent (realised I need to update some components I've written to do this, too!)

Only exception would be a `props` prop, which would handle any "other" props (see the Hamburger component for an example)

**Review by chriddell — CHANGES_REQUESTED** [review]
I'm also gonna pull this down and fix the rem() mixin before we merge

File: src/components/ResultItem/ResultItem.tsx (hunk 1 above)
Comment (DipakShrestha18) [reply src/components/ResultItem/ResultItem.tsx]: Done.

File: src/components/ResultItem/_resultItem.scss [hunk 2]
```diff
@@ -0,0 +1,84 @@
+@mixin font-size-with-line-height($fontSize, $lineHeightPx) {
```
Comment (chriddell) [inline src/components/ResultItem/_resultItem.scss]: filename should be kebab case (`result-item.scss`)

**Review by chriddell — CHANGES_REQUESTED** [review]

File: src/components/ResultItem/_resultItem.scss (hunk 2 above)
Comment (DipakShrestha18) [reply src/components/ResultItem/_resultItem.scss]: done.

File: src/assets/styles/30-components/__manifest.scss [hunk 3]
```diff
@@ -13,3 +13,5 @@
 @import './src/components/Nav/nav-list';
 @import './src/components/SearchPane/search-pane';
 @import './src/components/WellcomeCollectionBanner/wellcome-collection-banner';
+@import './src/Components/Logo/Logo';
+@import './src/components/ResultItem/ResultItem';
```
Comment (chriddell) [inline src/assets/styles/30-components/__manifest.scss]: path needs updating due to filename change

**Review by chriddell — CHANGES_REQUESTED** [review]

File: src/assets/styles/30-components/__manifest.scss (hunk 3 above)
Comment (DipakShrestha18) [reply src/assets/styles/30-components/__manifest.scss]: done.

**Review by chriddell — COMMENTED** [review]
All looks good except styles - I'm gonna work on them as would be great to have done as part of this PR - so likely have to wait til Monday to merge this.

**Comment by DipakShrestha18** [comment]
No probs.

**Review by chriddell — APPROVED** [review]

**Pull request merged**